superset/databases/schemas.py

```diff
@@ -254,6 +255,10 @@ def build_sqlalchemy_uri(
         the constructed SQLAlchemy URI to be passed.
         """
         parameters = data.pop("parameters", {})
+
+        # encode parameters
+        parameters = encode_parameters(parameters)
```

Comment:
We also need to add this to DB engine specs that implement their own `build_sqlalchemy_uri`.

Reply:
i added the encode_params to snowflake but, for gsheets and bigquery we don't leverage parameters to build the uri.
gsheets we always just return `"gsheets://` and for biguery we just leverage `encrypted_extra`